actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -252,6 +252,10 @@ def require(key)
     # Alias of #require.
     alias :required :require
 
+    def require_all(keys)
```

Comment:
The issue is that in Strong Parameters, `permit` is the heart of the logic that marks the attribute as whitelisted for future use in models. `require` simply will raise if not provided value, and AFAIK is mostly a glorified `presence` check before the `permit`.

If using `require` on individual attributes, you'd *still* need to `permit` them as well, or they will not be accepted by model (the most common use case), and `require` doesn't automatically `permit` the attribute.